dotCMS/src/main/java/com/dotcms/ema/proxy/ProxyTool.java

```diff
@@ -0,0 +1,1214 @@
+     * @param data - The String Data
+     * @param method - The Method (POST/PUT)
+     * @param creds - A credentials object for authenticated requests.
+     * @return A ProxyResponse Object
+     */
+    public ProxyResponse sendStringData(String url, String data, String method, Credentials creds) {
+        return sendStringData(url, data, method, DEFAULT_CONTENT_TYPE, creds);
+    }
+
+    /**
+     * This is used to send a single string payload to a url. Can be used to send JSON or XML to a url.
+     * Only supports PUT and POST Methods.
+     *
+     * @param url - The URL
+     * @param data - The String Data
+     * @param method - The Method (POST/PUT)
+     * @return A ProxyResponse Object
+     */
+    public ProxyResponse sendStringData(String url, String data, String method) {
+        return sendStringData(url, data, method, DEFAULT_CONTENT_TYPE, null);
+    }
+
+    /**
+     * This is used to post a single string payload to a url. Can be used to send JSON or XML to a url.
+     *
+     * @param url - The URL
+     * @param data - The String Data
+     * @param creds - A credentials object for authenticated requests.
+     * @return A ProxyResponse Object
+     */
+    public ProxyResponse sendStringData(String url, String data, Credentials creds) {
+        return sendStringData(url, data, METHOD_POST, DEFAULT_CONTENT_TYPE, creds);
+    }
+
+    /**
+     * This is used to post a single string payload to a url. Can be used to send JSON or XML to a url.
+     *
+     * @param url - The URL
+     * @param data - The String Data
+     * @return A ProxyResponse Object
+     */
+    public ProxyResponse sendStringData(String url, String data) {
+        return sendStringData(url, data, METHOD_POST, DEFAULT_CONTENT_TYPE, null);
+    }
+
+    /**
+     * Sends a request to a url.
+     *
+     * @param url - The URL
+     * @param params - The Query String
+     * @param method - The Method (POST/GET/PUT/HEAD/DELETE)
+     * @param contentType - The content type for this request
+     * @param creds - A credentials object for authenticated requests.
+     * @return A ProxyResponse Object
+     */
+    public ProxyResponse send(String url, Map<String, String> params, String method, String contentType, Credentials creds) {
+        Logger.debug(this, "send(Map) called with url=" + url + ", params=" + params + ", and method=" + method + ", contentType = "
+                + contentType + ", creds=" + UtilMethods.isSet(creds));
+
+
+        try {
+            List<NameValuePair> data = new ArrayList<NameValuePair>();
+            StringBuilder urlParamsSB = new StringBuilder();
+            String appender = "?";
+            if (url.contains("?")) {
+                appender = "&";
+            }
+            for (Entry<String, String> e : params.entrySet()) {
+                data.add(new BasicNameValuePair(e.getKey(), e.getValue()));
+                urlParamsSB.append(appender + e.getKey() + "=" + e.getValue());
```

Comment:
Codacy found an issue: [Avoid concatenating nonliterals in a StringBuffer/StringBuilder constructor or append().](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5922984)